### ISSUE TITLE
1.2.2 DNS Issues?

### ISSUE THREAD

**Issue by barqers** [hassio-addons/addon-node-red#97]
There's an error in 1.2.2 where the configuration nodes continually ask the user to login via the web GUI in node-red, and in the startup logs there's an error checking the password against the haveibeenpwned database = assumes password is safe and continues.

Likewise, the issue causes npm_packages passed through in the config to fail to install throwing the EAI_AGAIN error.

These 3 things point to a DNS error with node-red. I'm not sure what logs I could post to help?

> [cont-init.d] 81-npm-packages.sh: executing... 
> npm ERR! code EAI_AGAIN
> npm ERR! errno EAI_AGAIN
> npm ERR! request to https://registry.npmjs.org/node-red-contrib-light-scheduler failed, reason: getaddrinfo EAI_AGAIN registry.npmjs.org:443
> npm ERR! A complete log of this run can be found in:
> npm ERR!     /root/.npm/_logs/2018-12-23T15_40_28_706Z-debug.log
> FATAL: Failed installing npm package node-red-contrib-light-scheduler
> [cont-init.d] 81-npm-packages.sh: exited 1.
> [cont-finish.d] executing container finish scripts...
> [cont-finish.d] 99-message.sh: executing... 
> -----------------------------------------------------------
>                 Oops! Something went wrong.
>  
>  We are so sorry, but something went terribly wrong when
>  starting or running this add-on.
>  
>  Be sure to check the log above, line by line, for hints.
> -----------------------------------------------------------
> [cont-finish.d] 99-message.sh: exited 0.
> [cont-finish.d] done.
> [s6-finish] syncing disks.
> [s6-finish] sending all processes the TERM signal.

**Comment by tjorim** [comment]
One thing that would help is to post the full logs (which should include some system info).

**Comment by barqers** [comment]
> One thing that would help is to post the full logs (which should include some system info).

I did that - I posted the log from the hassio panel - is there a way I can get a more detailed log? I'm not sure how.

**Comment by barqers** [comment]
> First of all, a couple of remarks!
> 
> 1. Please, just use the template when filing an issue.
> 2. Use a code block for things like logs, not a quote.
> 
> (Re)start the add-on, click the 'refresh' button when the add-on is started up or stops. Then it should have the full logs (otherwise it only shows the last x lines I noticed).
> 
> We would need the things like Host OS (version and architecture), the HA and supervisor version...

Sorry about that - Let me try again:

# Problem/Motivation

> 1.2.2 Add-on for Node-red has DNS/authentication issues with Hassio 0.84.6

## Expected behavior

> Connects as previously done in 1.2.1

## Actual behavior

> Throws errors if "npm_packages" has nodes such as node-red-contrib-light-scheduler requested, logs into web GUI just fine but editing a home assistant node throws 'unauthorized' errors request login information again.

## Steps to reproduce

> Update to 1.2.2

## Logs
```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 00-banner.sh: executing... 
-----------------------------------------------------------
 Hass.io Add-on: Node-RED v1.2.2
 Flow-based programming for the Internet of Things
 From: Community Hass.io Add-ons
 By: Franck Nijhof <frenck@addons.community>
-----------------------------------------------------------
 armhf / HassOS 1.13 / HA 0.84.6 / SU 141 / stable
-----------------------------------------------------------
[cont-init.d] 00-banner.sh: exited 0.
[cont-init.d] 01-log-level.sh: executing... 
Log level is set to INFO
[cont-init.d] 01-log-level.sh: exited 0.
[cont-init.d] 02-updates.sh: executing... 
INFO: You are running the latest version of this add-on
[cont-init.d] 02-updates.sh: exited 0.
[cont-init.d] 10-requirements.sh: executing... 
curl: (6) Could not resolve host: api.pwnedpasswords.com
ERROR: Something went wrong contacting the HIBP API
WARNING: Could not check password, assuming it is safe.
curl: (6) Could not resolve host: api.pwnedpasswords.com
ERROR: Something went wrong contacting the HIBP API
WARNING: Could not check password, assuming it is safe.
[cont-init.d] 10-requirements.sh: exited 0.
[cont-init.d] 20-config.sh: executing... 
[cont-init.d] 20-config.sh: exited 0.
[cont-init.d] 50-remove-conflicts.sh: executing... 
up to date in 7.416s
[cont-init.d] 50-remove-conflicts.sh: exited 0.
[cont-init.d] 80-system-packages.sh: executing... 
[cont-init.d] 80-system-packages.sh: exited 0.
[cont-init.d] 81-npm-packages.sh: executing... 
npm ERR! code EAI_AGAIN
npm ERR! errno EAI_AGAIN
npm ERR! request to https://registry.npmjs.org/node-red-contrib-light-scheduler failed, reason: getaddrinfo EAI_AGAIN registry.npmjs.org:443
npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2018-12-23T16_36_35_669Z-debug.log
FATAL: Failed installing npm package node-red-contrib-light-scheduler
[cont-init.d] 81-npm-packages.sh: exited 1.
[cont-finish.d] executing container finish scripts...
[cont-finish.d] 99-message.sh: executing... 
-----------------------------------------------------------
                Oops! Something went wrong.
 
 We are so sorry, but something went terribly wrong when
 starting or running this add-on.
 
 Be sure to check the log above, line by line, for hints.
-----------------------------------------------------------
[cont-finish.d] 99-message.sh: exited 0.
[cont-finish.d] done.
[s6-finish] syncing disks.
[s6-finish] sending all processes the TERM signal.
```

**Comment by barqers** [comment]
> Are you using pi-hole?

Hey Frenck you're right I am using pi-hole - I disabled and changed my router settings. Is it not recommended to use in conjunction with pihole? I don't believe it's blocking any of the node-red servers since I'm only using the default pihole list at the moment.

**Comment by tjorim** [comment]
> Is it not recommended to use in conjunction with pihole?

You can use it but need to make sure that "your Hass.io machine [has] a static IP" (and DNS server in my experience), definitely when you set the DNS servers on the router instead of per device.

**Comment by frenck** [comment]
Yep, as @tjorim mentioned, ensure you have a static IP with a static (outside) DNS server configured on the device running Hass.io.

**Comment by barqers** [comment]
> Yep, as @tjorim mentioned, ensure you have a static IP with a static (outside) DNS server configured on the device running Hass.io.

Sorry let's say DNS server is test.duckdns.org and my IP for my pi is static at 192.168.0.120 - what would I still be missing? Because I have a static IP and a static outside DNS server setup?
